### PR TITLE
Block Removal of obsolete Code regarding old rendering

### PR DESCRIPTION
**Pull request by fhelfer**
Since old getHTML has been abandoned and force switched to getHTMLNew() for all Blocks, the old code fragments scattered in various components can be deleted.

### CODE REVIEW

**Comment by Amstutz** [comment]
I love this part:
image

**Comment by nicoroeser** [comment]
Thank you, @fhelfer, for your work on this! And thanks for putting the changes to the file headers in a separate commit. :smiley:

Let’s take this pull request further and prevent it from gathering dust. So far, I have assigned everyone as a reviewer who is responsible for changed code. Please have a look and give your go or raise objections.

Remember that the changes will most probably be also needed on trunk; tree structure has changed a bit there.

Anyone who likes to take the pull request on and merge the changes after the reviews?

Even though I don’t claim to understand all that’s going on here, just two comments from me:

1. While I understand what the change to `Services/Dashboard/ItemsBlock/classes/class.ilDashboardSidePanelSettingsRepository.php` does, I cannot directly see _why_ it is introduced. In case it is not directly related to the rest of the commit, I’d recommend to split the part out and explain it in the commit message.
2. `Services/Dashboard/classes/class.ilDashboardContentBlockGUI.php` still contains a function `fillDataSection`, which I _think_ is not used anymore. Is this an oversight or is the function required?

Fellow developers will know far better than me whether these things are issues or not.

**Review by iszmais — APPROVED** [review]
Dashboard LGTM.
Thanks a lot @fhelfer!
I highly appreciate such refactoring approaches.

Greetings,
@iszmais

**Review by alex40724 — APPROVED** [review]
Thanks for the PR! I just tested it and all my blocks are still working. Please note that I can't approve changes under Services/Block anymore. It's currently lacking an authority.

**Review by kergomard — APPROVED** [review]
Thank you very much @fhelfer for the PR. I approve the changes in `ilObject`.

**Comment by iszmais** [comment]
Hello,

> Let’s take this pull request further and prevent it from gathering dust. 

To clean off the dust of this, I would encourage all affected authorities to give their approval for their components (if not already happened). Be aware that your approval is only necessary for changes inside your component and not the entire PR:

Therefore, the following approval is still open:
- @schmitz-ilias for Poll
- @klees for StudyProgramme
- @mjansenDatabay for Mail
- @smeyer-ilias for Search

Further, I want to encourage any authority to take a look at one (or all) of the components without authority and give their approval:
- Block
- Container
- InfoScreen

I will happily await any feedback on those components until 29.08.2024 to give any interest enough time to consider a participation.
To keep the ball rolling after that, @gvollbach and I will take care of this PR then.

Greetings,
@iszmais

**Review by mjansenDatabay — APPROVED** [review]

**Review by klees — APPROVED** [review]

**Review by schmitz-ilias — APPROVED** [review]
Thanks for the PR! Changes in Poll look good, but please take out the duplicate `declare(strict_types=1);` in `ilPollBlockGUI` (just a small thing, so no need for a re-review...)

File: Services/Calendar/classes/class.ilCalendarBlockGUI.php
Comment (smeyer-ilias) [inline Services/Calendar/classes/class.ilCalendarBlockGUI.php:274]: Please also remove the template tpl.pd_event_list.html It is not used elsewhere.

**Review by smeyer-ilias — APPROVED** [review]
Thanks @iszmais for the PR.
The changes in Calendar and Search are fine for me.

**Comment by iszmais** [comment]
Thx, @smeyer-ilias but the credit actually goes to @fhelfer :D

@fhelfer can you adapt the last small change requests (Poll: dublicated type strict and Calendar: remove template). With that im confident this can be merged as soons as a shepherds is set.

Greetings,
@iszmais

**Comment by mjansenDatabay** [comment]
@iszmais / @fhelfer I guess further templates can be removed, not only `tpl.pd_event_list.html`.

File: Services/Mail/classes/class.ilPDMailBlockGUI.php
Comment (mjansenDatabay) [inline Services/Mail/classes/class.ilPDMailBlockGUI.php:126]: This template can be removed as well.

File: Services/Tasks/classes/class.ilPDTasksBlockGUI.php
Comment (mjansenDatabay) [inline Services/Tasks/classes/class.ilPDTasksBlockGUI.php:102]: I guess this template can re removed as well.

**Comment by iszmais** [comment]
Thank you a lot for your contribution!

This looks good to me and therefore has my approval (for the overall PR).
@gvollbach Pls add your review. As soon as this is done I will merge this into 9 and trunk.

Greetings,
@iszmais

**Review by iszmais — APPROVED** [review]

**Review by gvollbach — APPROVED** [review]
@iszmais Looks good to me, thx.

**Comment by iszmais** [comment]
I merged this into 9, 10 and trunk.

@fhelfer Thank you again for your great contribution!

Such refactorings might be tedious and not as "exciting" as innovative features, but are as crucial to the core as any other development.

Greetings,
@iszmais

**Pull request merged**
